### PR TITLE
feat: Check the interface ids of the LSPs on the inspect page.

### LINKED ISSUE #4: Use interfaceId to detect the type of LSP contract
List here:
<https://docs.lukso.tech/standards/smart-contracts/interface-ids>

### PR DESCRIPTION
**Pull request by b00ste**
## What does this PR introduce?

This PR checks for the interface id of the LSP contracts and displays if the address is a valid LSP contract.

## Contracts tested:

UniversalProfile: `0x940EAC4bE26A42884883A79b6ee7179a24c42792`
LSP6KeyManager: `0x75b0B516B47A27b1819D21B26203Abf314d42CCE`
LSP1UniversalReceiverDelegateUP: `0x906B067e392e2c5f9E4f101f36C0b8CdA4885EBf`
LSP1UniversalReceiverDelegateVault: `0xDf951d2061b12922BFbF22cb17B17f3b39183570`
LSP7Mintable: `0xe14058B1c3def306e2cb37535647A04De03Db092`
LSP8Mintable: `0xF5b81Fe0B6F378f9E6A3fb6A6cD1921FCeA11799`
LSP9Vault: `0x67baFF31318638F497f4c4894Cd73918563942c8`
UniversalProfileInit: `0x6533158b042775e2FdFeF3cA1a782EFDbB8EB9b1`
LSP6KeyManagerInit: `0x73C68f1f41e4890D06Ba3e71b9E9DfA555f1fb46`
LSP7MintableInit: `0xD2D5e508C82EFc205cAFA4Ad969a4395Babce026`
LSP8MintableInit: `0x2b639Cc84e1Ad3aA92D4Ee7d2755A6ABEf300D72`
LSP9VaultInit: `0xF85895D097B2C25946BB95C4d11E2F3c035F8f0C`

Closes #4

### CODE REVIEW

**Comment by Hugoo** [comment]
The idea is to do this:
https://github.com/ERC725Alliance/erc725.js/issues/205

@fhildeb is on it atm - then we can use that new function.

**Review by Hugoo — APPROVED** [review]

File: interfaces/interface_ids.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const LSP_INTERFACE_IDS = {
```
Comment (CJ42) [inline interfaces/interface_ids.ts:1]: Why not using the interface IDs of the lsp-smart-contracts package? Might be easier to maintain.

File: pages/inspect.tsx [hunk 2]
```diff
@@ -95,15 +120,35 @@ const Home: NextPage = () => {
...
+          <p>LSP7DigitalAsset: ❌</p>
+          <p>LSP8IdentifiableDigitalAsset: ❌</p>
+          <p>LSP9Vault: ❌</p>
           <p>
-            This address is not a valid ERC725 Profile. Please check the address
-            is correct.
+            This address is not a valid ERC725 Profile, nor it is a valid LSP
+            contract.
           </p>
+          <p>Please check if the addressis correct.</p>
```
Comment (CJ42) [inline pages/inspect.tsx:151]: Added review comment

**Review by CJ42 — COMMENTED** [review]
Added review comment

File: interfaces/interface_ids.ts (hunk 1 above)
Comment (Hugoo) [reply interfaces/interface_ids.ts:1]: Yes this is a good point indeed

**Comment by Hugoo** [comment]
I suggest to put this on the right column as it starts to be a bit big:

image

Also, I think we can add a link from each LSPX... to the corresponding docs page.

**Pull request merged**